gatherer/src/main/java/de/zalando/pgobserver/gatherer/ConfigChecker.java

```diff
@@ -28,15 +29,13 @@ public void run() {
         while (true) {
 
             try {
-                Thread.sleep(CONFIG_CHECK_INTERVAL_SECONDS * 1000);
-
                 Map<Integer, Host> hosts_new = Host.LoadAllHosts(config);
 
                 applyConfigChangesIfAny(hosts_new);
 
-                LOG.error("finished checking new config settings. sleeping for {} s", CONFIG_CHECK_INTERVAL_SECONDS);
-            } catch (InterruptedException ie) {
-                LOG.error("", ie);
+                LOG.info("finished checking new config settings. sleeping for {} s", CONFIG_CHECK_INTERVAL_SECONDS);
```

Comment:
Ok, you are using scheduler: https://github.com/zalando/PGObserver/pull/78/files#diff-7904d0d631bdf6ac4e13aa3cdfb60b8dR89

But in this case you probably don't need to have infinite loop: https://github.com/zalando/PGObserver/pull/78/files#diff-bec9f7e12ea896cc43dd950e0736e664R29